# HG changeset patch
# User Rik <rik@octave.org>
# Date 1452790304 28800
#      Thu Jan 14 08:51:44 2016 -0800
# Node ID a1aadf619e3c4fe8ffbb64e06d31662b2eae9011
# Parent  c221ce56f774983775e1c9366b80b6fc60edf7e5
inputname.m: Clarify docstring.

* inputname.m: Clarify docstring.  Rename output variable 's' to 'name' for
clarity.

diff --git a/scripts/miscellaneous/inputname.m b/scripts/miscellaneous/inputname.m
--- a/scripts/miscellaneous/inputname.m
+++ b/scripts/miscellaneous/inputname.m
@@ -18,39 +18,42 @@
 ##
 ## Original version by Paul Kienzle distributed as free software in the
 ## public domain.
 
 ## -*- texinfo -*-
 ## @deftypefn {} {} inputname (@var{n})
 ## Return the name of the @var{n}-th argument to the calling function.
 ##
-## If the argument is not a simple variable name, return an empty string.
-## @code{inputname} may only be used within a function body, not at the
-## command line.
+## If the argument is not a simple variable name, return an empty string.  As
+## an example, a reference to a field in a structure such as @code{s.field} is
+## not a simple name and will return @qcode{""}.
+##
+## @code{inputname} is only useful within a function.  When used at the command
+## line it always returns an empty string.
 ## @seealso{nargin, nthargout}
 ## @end deftypefn
 
-function s = inputname (n)
+function name = inputname (n)
 
   if (nargin != 1)
     print_usage ();
   endif
 
-  s = "";
+  name = "";
   try
-    s = evalin ("caller", sprintf ("__varval__ ('.argn.'){%d}", fix (n)));
+    name = evalin ("caller", sprintf ("__varval__ ('.argn.'){%d}", fix (n)));
   catch
     return;
   end_try_catch
 
   ## For compatibility with Matlab,
   ## return empty string if argument name is not a valid identifier.
-  if (! isvarname (s))
-    s = "";
+  if (! isvarname (name))
+    name = "";
   endif
 
 endfunction
 
 
 ## Warning: heap big magic in the following tests!!!
 ## The test function builds a private context for each test, with only the
 ## specified values shared between them.  It does this using the following
